### PR TITLE
[WIP] Include document describing UI flow

### PR DESCRIPTION
**Pull request by nmearl**
This document describes a loose formalism for implementing the front-end glue-jupyter ui framework using ipyvuetify. It is currently a work in progress.

### CODE REVIEW

File: docs/ui_description.rst [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+model (e.g. the core elements from Glue that are not dependent on the
+front-end library), and the view of the data (in this case, web-based
+tooling provided by Jupyter).
+
+Glupyter implements a base ``JupyterApplication`` object through which
+users can manage their data, create viewers, and add links between data
+sets. The data management and linking are controlled separate from the
+viewers in that changes made directly to the data state propagate to the
+UI (i.e. the UI acts strictly as the view to the data state, not the
+controller of the data state). The viewers themselves are based on the
```
Comment (astrofrog) [inline docs/ui_description.rst]: The 'not the controller of the data state' might need to be explained a bit more, in that we can still change the data/model state from the UI.

File: docs/ui_description.rst [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+IPyWidget package which allows the creation of widgets that can be used
```
Comment (astrofrog) [inline docs/ui_description.rst]: I think it's just 'ipywidgets' in terms of capitalization - also maybe add a link?

File: docs/ui_description.rst [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+and interacted with in python, but rendered in a javascript environment.
+
+There are two distinct use cases for the glupyter environment: first, as
+a means to procedurally interact with pieces of a user's workflow *in
+addition* to their work in e.g. a Jupyter Notebook; and second, to
+provide users a web-based gui to interact with and visualize their data
```
Comment (astrofrog) [inline docs/ui_description.rst]: gui -> GUI (occurs in other places too)

File: docs/ui_description.rst [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+while hiding the Jupyter context, e.g. a standalone web application.
```
Comment (astrofrog) [inline docs/ui_description.rst]: hiding the Python context? (i.e. it's more about hiding Python code than Jupyter?)

File: docs/ui_description.rst [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+These two use cases describe a data-first and gui-first approach,
+respectively. This document will focus on detailing the design of the
+gui-first approach.
+
+General user interface design
+-----------------------------
+
+The general user interface is a parallel design to the Qt desktop
+interface. It is meant to be the standard scaffolding supporting the
+implementation of the individual viewers, data management functions, and
```
Comment (astrofrog) [inline docs/ui_description.rst]: implementation and display of?

To me, the general UI doesn't really support implementation, it's more about putting the pieces together?

File: docs/ui_description.rst [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+user plugins in much the same way that the Qt desktop version does.
+
+The implementation leverages three primary packages:
+
+1. ``glue-jupyter``: handles the data and state management, including
+   the plugin infrastructure that provides the registry of available
+   viewers, analysis functions, etc.
+2. ``ipyvuetify``: provides the ui widgets for composing the web-based
```
Comment (astrofrog) [inline docs/ui_description.rst]: ui -> UI

File: docs/ui_description.rst [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+   front-end.
+3. ``ipygoldenlayout``: an additional widget that supports tabbing and
```
Comment (astrofrog) [inline docs/ui_description.rst]: what about making the package names links instead of fixed width names?

**Comment by nmearl** [comment]
Moved to [JDAViz](https://github.com/spacetelescope/jdaviz/pull/48).

**Pull request closed without merging**